direnvrc

```diff
@@ -27,26 +27,28 @@ _nix_direnv_fatal() {
   exit 1
 }
 
+_nix() {
+  nix --extra-experimental-features "nix-command flakes" "$@"
+}
+
 _nix_direnv_preflight () {
   if [[ -z "$direnv" ]]; then
     _nix_direnv_fatal "\$direnv environment variable was not defined. Was this script run inside direnv?"
   fi
 
-
-  if [[ -z ${NIX_BIN_PREFIX:-} ]]; then
-    NIX_BIN_PREFIX=$(command -v nix-shell)
-    if [[ -z "${NIX_BIN_PREFIX}" ]]; then
-      _nix_direnv_fatal "command not found: nix-shell."
-    fi
-    NIX_BIN_PREFIX="${NIX_BIN_PREFIX%/*}/"
+  if ! has direnv_version || ! direnv_version "$REQUIRED_DIRENV_VERSION" 2>/dev/null; then
+    _nix_direnv_fatal "base direnv version is older than the required v$REQUIRED_DIRENV_VERSION."
   fi
 
+  if [[ -n ${NIX_BIN_PREFIX:-} ]]; then
+    PATH_add "$NIX_BIN_PREFIX"
```

Comment:
maybe we can append NIX_BIN_PREFIX later on... so that if the user has nix in PATH it will be used.